Upgrade Asset Modules

PiyushChandra17 opened the pull request:
Fixes #2023 

Changes:

I have verified that this pull request:

* [x] has no linting errors (`npm run lint`)
* [x] has no test errors (`npm run test`)
* [x] is from a uniquely-named feature branch and is up to date with the  `develop` branch.
* [x] is descriptively named and links to an issue number, i.e. `Fixes #123`

PiyushChandra17 committed:
Upgrade Asset Modules

raclim commented:
Thanks for your work on this! Since there's currently an open PR (https://github.com/processing/p5.js-web-editor/pull/2957) for this issue at the moment, I think we might stick with that one to resolve it. I'm sorry that we couldn't get this one in, but will check out the other ones you've raised!

PiyushChandra17 commented:
@raclim I think this PR is `up to date one` with context with the latest webpack version, also i have gone through the webpack docs [here](https://webpack.js.org/guides/asset-modules/). 

Made the following changes:
+ npm installed latest webpack version
+ npm uninstalled `file-loader`, `raw-loader`
+ replaced `file-loader` with `asset/resource`
+ replaced `raw-loader` with `asset/source`
+ added generator following the docs
+ Used hash version in production, i think that's what the docs states

PS: If this goes well, i can reopen this PR in future i guess else we can stick the existing one
